File: SWProgrammingCompetencyTest/SWProgrammingCompetencyTest5/Docs/Q5_solution.java
import drill.test.*;

public class DrillTest {

	public int[][] testParameters = {
			{10, 2, 30, 4, 1},
			{30, 4, 10, 2, 0},	
			{10, 4, 20, 5, 0},	
			{5, 4, 30, 4, 0},		
			{10, 2, 45, 4, 0},	
			{10, 6, 30, 4, 0},	
			{10, 2, 30, 6, 0},	
			{10, 0, 30, 4, 0},		
			{10, 2, 30, 0, 0},	
			{50, 2, 20, 4, 0},	
			{10, 2, 50, 4, 0},	

	};

	public void runAll() {
		for (int i = 0; i < testParameters.length; ++i) {
			System.out.print(String.format("[%d] ", i));
			System.out.print(String.format("Expected = %d, ", testParameters[i][4]));
			int result = Machine.drill(testParameters[i][0], testParameters[i][1], testParameters[i][2], testParameters[i][3]);
			System.out.println(String.format("Return = %d", result));
		}
		
	}
	
	public static void main(String[] args) {
		DrillTest test = new DrillTest();
		test.runAll();
	}


}

File: SWProgrammingCompetencyTest/SWProgrammingCompetencyTest5/Docs/Q5_solution.cpp
#include <iostream>

#include "drill.h"

using namespace std;

int tarr[][5] = { 
	{10, 0, 30, 4, 0 },
	{10, 2, 20, 0, 0 }, 
	{30, 6, 46, 1, 0},	
	{20, 2, 36, 6, 0},  
	{10, 1, 11, 1, 0 },	
	{0, 2, 11, 2, 0 }, 
	{10, 2, 50, 2, 0 }, 
	{20, 2, 10, 2, 0 } 
};

int main()
{
	int i;

	for(i = 0; i <(sizeof(tarr)/sizeof(tarr[0])); i++)
	{
		cout << "[" << i << "]";
		cout << " Expected = " << tarr[i][4];
		cout << " Return = " << drill(tarr[i][0], tarr[i][1], tarr[i][2], tarr[i][3]);
		cout << endl;
	}

	return 0;
}
